### PR TITLE
plugin/metrics: support HTTPS qType in requests count metric label

### PR DESCRIPTION
**Pull request by plieskovsky**
### 1. Why is this pull request needed and what does it do?
Since iOS14, apple devices are standardly using the HTTPS query (type 65). It would be beneficial to be able to distinguish the amount of these requests based on the metrics. Therefore I would like to add the HTTPS query type support into the metric labels.

### 2. Which issues (if any) are related?
none

### 3. Which documentation changes (if any) need to be made?
none

### 4. Does this introduce a backward incompatible change or deprecation?
no

### CODE REVIEW

**Comment by SuperQ** [comment]
This needs a DCO sign-off. You can use `git commit -s --amend` to add it.

**Comment by plieskovsky** [comment]
@SuperQ added, thank you for the info 👍

**Review by SuperQ — APPROVED** [review]
LGTM

**Comment by miekg** [comment]
happy to add this, but isn't the HTTPS still in draft? Might be prudent to wait until RFC.

yeah it still draft: https://datatracker.ietf.org/doc/draft-ietf-dnsop-svcb-https/

**Comment by plieskovsky** [comment]
@miekg ah, that's interesting site/fact. Didn't know about it.
On the other hand, iOS has been already using it for about 4 months. We can see a non-trivial amount (~18%) of HTTPS requests sent to our CoreDNS instance that faces the internet. So I guess we cannot go wrong with enabling the computation of it into the metric label.

But as you guys are the owners/maintainers of the repo, the decision is totally up to you :)

**Comment by miekg** [comment]
[ Quoting ***@***.***> in "Re: [coredns/coredns] plugin/metric..." ]
***@***.*** ah, that's interesting site/fact. Didn't know about it.
>On the other hand, iOS has been already using it for about 4 months. We can see
>a non-trivial amount (~18%) of HTTPS requests sent to our CoreDNS instance that
>faces the internet. So I guess we cannot go wrong with enabling the computation
>of it into the metric label.
>
>But as you guys are the owners/maintainers of the repo, the decision is totally
>up to you :)

yeah, it's a label value so easily changed and I can see this RR being import, but it
might still see a name change. SVCB might be nice as well (same caveats)

/lgtm
/merge

**Pull request merged**
